File: model/src/main/java/io/github/runtimemodels/chazm/parsers/ParsersModule.java
package io.github.runtimemodels.chazm.parsers;

import com.google.inject.AbstractModule;
import io.github.runtimemodels.chazm.OrganizationModule;

/**
 * The {@linkplain ParsersModule} class provides a Guice binding module for parsers.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
class ParsersModule extends AbstractModule {

    @Override
    protected void configure() {
        bind(XmlParser.class);
        install(new OrganizationModule());
    }

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/guice/ParsersModule.kt
package runtimemodels.chazm.model.guice

import com.google.inject.AbstractModule
import com.google.inject.Provides
import runtimemodels.chazm.model.parser.XmlParser
import javax.xml.stream.XMLInputFactory

/**
 * The [ParsersModule] class provides a Guice binding module for parser.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
internal class ParsersModule : AbstractModule() {

    @Provides
    fun xmlInputFactory(): XMLInputFactory = javax.xml.stream.XMLInputFactory.newInstance()

    override fun configure() {
        bind(XmlParser::class.java)

        install(OrganizationModule())
    }

}
